zerver/models.py

```diff
@@ -1339,21 +1316,13 @@ def linkifiers_for_realm_remote_cache(realm_id: int) -> List[LinkifierDict]:
 def flush_linkifiers(*, instance: RealmFilter, **kwargs: object) -> None:
     realm_id = instance.realm_id
     cache_delete(get_linkifiers_cache_key(realm_id))
-    with suppress(KeyError):
-        per_request_linkifiers_cache.pop(realm_id)
+    flush_per_request_cache("linkifiers_for_realm")
```

Comment:
Doesn't this remove everything from that cache? It's perhaps super important, since linkifiers are rarely edited in any case, but there's no reason to flush other realm's linkifiers due to an edit to just this one realm.

Reply:
Oh but that's irrelevant, because each request only accesses one realm's linkifiers anyway. OK.

Reply:
Yeah, I explained it in the commit message.  It's not a perfect long-term decision, but hopefully this code will shrink more than it grows.